app/src/main/java/com/example/dine_and_donate/Adapters/EventListViewAdapter.java

```diff
@@ -0,0 +1,87 @@
+package com.example.dine_and_donate.Adapters;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.bumptech.glide.Glide;
+import com.example.dine_and_donate.Models.Event;
+import com.example.dine_and_donate.Models.User;
+import com.example.dine_and_donate.R;
+import com.google.firebase.database.DataSnapshot;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class EventListViewAdapter extends RecyclerView.Adapter<EventListViewAdapter.ViewHolder> {
+
+    static Context context;
+    private ArrayList<Event> mEvents = new ArrayList<>();
+    private HashMap<String, JSONObject> mIdToRestaurant;
+    private HashMap<String, User> mIdToOrg;
+
+    public EventListViewAdapter(DataSnapshot allEvents, HashMap<String, JSONObject> mIdToRestaurant, HashMap<String, User> mIdToOrg) {
+        for(DataSnapshot ds : allEvents.getChildren()) {
+            mEvents.addAll(Event.eventsHappeningAtRestaurant(ds));
+        }
+        this.mIdToRestaurant = mIdToRestaurant;
+        this.mIdToOrg = mIdToOrg;
+    }
+
+    @NonNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        context = parent.getContext();
+        LayoutInflater inflater = LayoutInflater.from(context);
+
+        View view = inflater.inflate(R.layout.layout_event_list_view, parent, false);
+        ViewHolder viewHolder = new ViewHolder(view);
+        return viewHolder;
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
+        Event event = mEvents.get(position);
+        holder.tvOrgName.setText(mIdToOrg.get(event.orgId).name);
+        holder.tvOrgInfo.setText(event.info);
+        try {
+            holder.tvRestaurantName.setText(mIdToRestaurant.get(event.yelpID).getString("name"));
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+        Glide.with(context)
+                .load("https://firebasestorage.googleapis.com/v0/b/dine-and-donate.appspot.com/o/images%2F158765210?alt=media&token=be40174f-ed03-4299-8431-410b036a9037")
```

Comment:
is this how you're supposed to do it?

Reply:
This is just a hardcoded value for now to fill the image view, I will change this to be the actual profile image soon